src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -159,19 +161,12 @@ export const getMaybeProvider = async (
   if (!isKeyManagerWithProvider(localWalletType)) return null
 
   if (localWalletType === KeyManager.MetaMask) {
+    // TODO(gomes): get by rdns and hasta la vista detectEthereumProvider()
```

Comment:
Not for this PR, since I didn't manage to remove it without regressions after a few tries, and it accidentally works fine without - TBD as a granular PR with fresh eyes.

List of TODOs:

 - [ ] Phantom rdns detection in hdwallet
 - [ ] move ugly metamask-snaps JSON-rpc bits over to web, so we don't have to fiddle with this monstrocity of a monorepo
 - [ ] Keplr EVM in hdwallet with rdns
 - [ ] cleanup accounts and chains listeners bs - we should only attach event listeners for the current rdns *if* connected
 - [ ] Do the two above with two birds one stone, leveraging shared utils

Reply:
And then this package will go out from the bundle fully :fridaydog:

Reply:
Actually done in this PR with `useEip1993EventHandler` 🎉 `detectEthereumProvider` is gone, the whole "remove event listeners when connecting another wallet" is gone since we can use the correct ones for the current rdns / first-class hdwallet (Phantom, Coinbase)